### PR TITLE
Fix E2E new checkout compatibility

### LINKED ISSUE #3092: All e2e tests fail for the updated experience (aka UPE) since 8.0.0
**Describe the bug**

When UPE is active, all of the e2e tests fail since 8.0.0. They pass for versions < 8.0.0, so this is surely related to the Split PE changes.

**To Reproduce**
1. Install [8.0.0](https://github.com/woocommerce/woocommerce-gateway-stripe/releases/tag/8.0.1) (or any more recent version) on a JN site
2. Ensure the Updated checkout experience is enabled
3. Run the e2e tests against this site, as explained [in the e2e readme](https://github.com/woocommerce/woocommerce-gateway-stripe/tree/develop/tests/e2e)
4. Notice all of the tests fail 

     

5. Install Stripe [7.9.1](https://github.com/woocommerce/woocommerce-gateway-stripe/releases/tag/7.9.1) on this JN site
6. Ensure the Updated checkout experience is enabled
7. Notice that all of the tests pass (except maybe the refund one with which we were having problems)

**Expected behavior**

e2e tests must pass when the Legacy checkout experience is disabled. 

**Additional context**

When handling payments, we have conditionals [like these](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/tests/e2e/utils/payments.js#L38) for checking whether UPE is enabled. 

With the introduction of Split PE in 8.0.0, the selectors we're looking for were most likely updated.

### PR DESCRIPTION
**Pull request by diegocurbelo**
Issue: https://github.com/woocommerce/woocommerce-gateway-stripe/issues/3092

## Changes proposed in this Pull Request:

To keep the needed changes manageable, this PR does not fix #3092; it just refactors the existing E2E setup and tests to split the execution into different projects.

Now the E2E test run follow this pattern:
- Disable legacy checkout experience (**default**)
- Run tests (all but those in the `_legacy-experience` subfolder), **(currently empty)**.
- Enable legacy checkout experience (**legacy-experience**)
- Run legacy tests (`_legacy-experience` subfolder)

Screenshot 2024-05-01 at 2 51 08 AM

**NOTE:** It also updates Playwright (needed for playwright project dependencies to work) and updates the browsers after running `npm install`.

## Testing instructions

Confirm all of the E2E tests  [running](https://github.com/woocommerce/woocommerce-gateway-stripe/actions/runs/8901802199/job/24446411368?pr=3106) in this PR pass:

Screenshot 2024-04-30 at 6 33 44 PM

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -5,7 +5,6 @@ services:
     build: ./docker/wordpress_xdebug
     image: wordpress-xdebug
     container_name: woocommerce_stripe_wordpress
-    restart: always
```
Comment (Mayisha) [inline docker-compose.yml:8]: Quick question, why are we removing the restart policy?

**Review by Mayisha — APPROVED** [review]
Thanks @diegocurbelo. The changes look good :+1: 

I have not run the e2e tests locally since all E2E tests in the GH checks are passing :sweat_smile:

**Review by a-danae — COMMENTED** [review]
Sweet! Thanks for reorganizing the tests, @diegocurbelo !

I'm getting the following when running the tests on a JN site. Maybe I'm missing something?

Ran `npm i` first. I'm using 8.2.0. Tried with Legacy enabled and disabled. This happens for me with both `npm run test:e2e-setup --` and `npm run test:e2e --`

| List of tests | Failing test |
|--------|--------|
| image | image |

File: docker-compose.yml (hunk 1 above)
Comment (diegocurbelo) [reply docker-compose.yml:8]: For consistency, only the WordPress container was started automatically after a reboot.
Another option would've been adding always to the DB container, but we don't usually want the containers to start after every reboot.

**Comment by diegocurbelo** [comment]
> Sweet! Thanks for reorganizing the tests, @diegocurbelo !
> 
> I'm getting the following when running the tests on a JN site. Maybe I'm missing something?
> 
> Ran `npm i` first. I'm using 8.2.0. Tried with Legacy enabled and disabled. This happens for me with both `npm run test:e2e-setup --` and `npm run test:e2e --`
> 
> List of tests	Failing test
> image	image

The new tests don't work with previous versions, as there was no hay before to make sure the correct checkout experience was enabled... you need to upload a zip from this branch.

Also, the Disable legacy experience step enabled the new checkout before running the tests... but I updated the PR to have it set as `YES` by default; to be consistent with a new default onboarding.

**Pull request merged**